### PR TITLE
Locations Incremental Sync Bug Fix

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

Previously, when incrementally syncing a Web User, the system did not check whether the set of currently assigned locations had changed. As a result, any additional locations assigned to a Web User were not correctly populated in the fixture during the incremental sync.

Fix Implemented: This PR addresses the issue by ensuring that any new locations assigned to a Web User are now correctly added to the fixture during incremental sync.

## Technical Summary

Jira Ticket: https://dimagi.atlassian.net/browse/USH-4764
The strategy implemented for WebUser is the same as that used for `CommCareUser`. It involves storing a timestamp to track any changes in a `WebUsers`'s assigned location. During incremental sync, this timestamp is compared against the last sync time to determine if any locations need to be synced.

## Feature Flag

No FF

### Safety story

Tested locally and on staging. The design and underlying logic has already been used for `CommCareUser`. This PR only expands to also encompass webUsers.

### Automated test coverage

test exists that checks `UserLookupTableStatus` is updated properly. This PR adds tests that change to `WebUser` locations also updates the `UserLookupTableStatus` model.

### QA Plan

no QA

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by AddisonDunn — COMMENTED** [review]
Current code looks good to me. I think you need to update `OTARestoreWebUser.get_fixture_last_modified` to see the timestamp take effect when syncing though.

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -2584,7 +2586,7 @@ def unset_location(self, domain, fall_back_to_next=False, commit=True):
         membership = self.get_domain_membership(domain)
         old_location_id = membership.location_id
         if old_location_id:
-            membership.assigned_location_ids.remove(old_location_id)
+            self._remove_location_from_user(membership, old_location_id)
```
[the remark this answers is not in the document]
Comment (Jtang-1) [reply corehq/apps/users/models.py:2589]: What do you mean by "removing the primary location (but not changing it)"? Agreed that I think the timestamp should be reset only on assigned locations changing. It looks like `unset_location` removes the primary location but also removes it from the `assigned_location_ids`.

**Comment by Jtang-1** [comment]
@AddisonDunn I updated `OTARestoreWebUser.get_fixture_last_modified` in this commit [e71bb5a](https://github.com/dimagi/commcare-hq/pull/34900/commits/e71bb5a81d7487d709b5281b19f7d11a4474b443). Is there something I missed in that change?

File: corehq/apps/users/tests/fixture_status.py [hunk 2]
```diff
@@ -58,38 +60,75 @@ def test_get_status(self):
...
+        couch_user_new_updated_time = UserLookupTableStatus.objects.get(user_id=self.couch_user._id).last_modified
+        self.assertTrue(couch_user_new_updated_time > couch_user_prev_updated_time)
+        web_user_new_updated_time = UserLookupTableStatus.objects.get(user_id=self.web_user._id).last_modified
+        self.assertTrue(web_user_new_updated_time > web_user_prev_updated_time)
+
+    def test_unset_location_by_id(self):
+        fake_location1 = MagicMock()
+        fake_location2 = MagicMock()
+        fake_location1.location_id = "misty_mountains"
+        fake_location2.location_id = "lonely_mountains"
```
[the remark this answers is not in the document]
Comment (Jtang-1) [reply corehq/apps/users/tests/fixture_status.py]: Vital to keep factually accurate to the lore haha [fcb4cab](https://github.com/dimagi/commcare-hq/pull/34900/commits/fcb4cab871e3710646b760132eb1f6311d4e50e1)

File: corehq/apps/users/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jtang-1) [reply corehq/apps/users/models.py:2589]: oh, I was looking at the `_remove_location_from_user` case. I agree, I don't know see why the timestamp should be reset in the line that you linked.

File: corehq/apps/users/tests/fixture_status.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinRiese) [reply corehq/apps/users/tests/fixture_status.py]: that made my morning

**Comment by Jtang-1** [comment]
@AddisonDunn I added a test that `get_fixture_last_modified` results in `_fixture_has_changed` and `should_sync_locations` evaluating as `True`  [f5ffc23](https://github.com/dimagi/commcare-hq/pull/34900/commits/f5ffc23da045cfb6b324f89139b525c3c5157258)

**Review by AddisonDunn — APPROVED** [review]
Looks good. Sorry @Jtang-1 , when I said "because it hasn't been tested yet" I meant not tested by delivery on staging, I wasn't asking for an automated test. But I love the test you added.

**Pull request merged**